### PR TITLE
Do note fetch cloud-config on unauthenticated registartion calls

### PR DESCRIPTION
**Pull request by davidcassany**
Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

File: pkg/apis/elemental.cattle.io/v1beta1/zz_generated_deepcopy.go [hunk 1]
```diff
@@ -22,7 +22,6 @@ limitations under the License.
 package v1beta1
 
 import (
-	config "github.com/rancher/elemental-operator/pkg/config"
```
Comment (davidcassany) [inline pkg/apis/elemental.cattle.io/v1beta1/zz_generated_deepcopy.go:25]: these changes are the result of calling `go generate`

File: pkg/server/register.go [hunk 2]
```diff
@@ -105,19 +105,13 @@ func (i *InventoryServer) ServeHTTP(resp http.ResponseWriter, req *http.Request)
 }
 
 func (i *InventoryServer) unauthenticatedResponse(machineRegistration *elm.MachineRegistration, writer io.Writer) error {
-	_, err := writer.Write([]byte("#cloud-config\n"))
```
Comment (davidcassany) [inline pkg/server/register.go:108]: This was never meant to be a cloud-config as described by [cloud-init docs](https://cloudinit.readthedocs.io/en/latest/topics/examples.html) neither a yip config file. I have no idea why this mark was included here at some point.

**Comment by Itxaka** [comment]
Should we actually return an http error 403? So it can be dealt with on the client? I did not understand why we were returning anything but a 403 on unauthorized requests....

File: pkg/server/register.go [hunk 3]
```diff
@@ -105,19 +105,13 @@ func (i *InventoryServer) ServeHTTP(resp http.ResponseWriter, req *http.Request)
...
-	}
-
 	return yaml.NewEncoder(writer).Encode(config.Config{
 		Elemental: config.Elemental{
 			Registration: config.Registration{
 				URL:    machineRegistration.Status.RegistrationURL,
 				CACert: i.getRancherCACert(),
 			},
 		},
-		CloudConfig: machineRegistration.Spec.Config.CloudConfig,
```
Comment (davidcassany) [inline pkg/server/register.go:120]: This should not be provided at this stage.

File: pkg/apis/elemental.cattle.io/v1beta1/zz_generated_deepcopy.go (hunk 1 above)
Comment (Itxaka) [reply pkg/apis/elemental.cattle.io/v1beta1/zz_generated_deepcopy.go:25]: This has me worried. I can seem to understand the relationship between your change and why was this auto-generated. I feel dumb

Comment (davidcassany) [reply pkg/apis/elemental.cattle.io/v1beta1/zz_generated_deepcopy.go:25]: There is no relation I believe we just missed to call the code generation on a previous PR. Probably you remember, I think I saw some time ago a repository that had a `validate` target on the make file that was actually checking if the generated code was introducing changes or not. I'd like to have a check in the CI the that tells me I should call `go generate` and commit again as the changes affect the generated code.

**Comment by davidcassany** [comment]
> Should we actually return an http error 403? So it can be dealt with on the client? I did not understand why we were returning anything but a 403 on unauthorized requests....

mmm I am not understanding the comment... 😅 This PR is to prevent the registration.yaml to include anything else than the registrationURL and the cacert. 

So with the following steps I noted the issue I am trying to solve:

1. Manualy created a machineregistration.yaml including some `cloud-config`
2. `kubectl apply -f manchineRegistration.yaml`
3. `kubectl get -n fleet-default machineregistrations.elemental.cattle.io test-nodes -o yaml` to get the registrationURL
4. `curl -k <registrationURL> > registration.yaml` this is the data to embed into the live ISO to automate elemental installation using `elemental-operator`

The problem is that the `registration.yaml` file now includes the cloud-config data which is not needed at this stage, as the node when it is authenticated fetches the full config including cloud-config and all installation details.

**Review by fgiudici — APPROVED** [review]
Yep, looks good here!

File: pkg/server/register.go (hunk 2 above)
Comment (fgiudici) [reply pkg/server/register.go:108]: yeah... nice cleanup

**Comment by Itxaka** [comment]
> The problem is that the registration.yaml file now includes the cloud-config data which is not needed at this stage,

Yes, on unauthorized calls right? So I'm wondering, if you call the registration URL and are not authorized, why aren't we even answering with anything? We should probably return an http 403 as good citizens do.

This is a good step, but I was wondering if it's even needed to answer anything at all.


Unless this is the response of the authorized call?

File: pkg/server/register.go (hunk 2 above)
Comment (mudler) [reply pkg/server/register.go:108]: I'm not 100% sure if there was some other intention behind to push some cloud config at those stages, but it looks good as the relevant parts we have to return are just part of the defined schema

**Review by mudler — APPROVED** [review]

**Comment by fgiudici** [comment]
> We should probably return an http 403 as good citizens do

Yeah, make sense... I would like to investigate more what happens when we register and things go wrong (right now I think the operator gets stuck with no logs). I think we should address this too in https://github.com/rancher/elemental-operator/issues/68

**Comment by Itxaka** [comment]
Yep, tpm.get is stuck retrying, no sure if forever, but for a while.

I believe there is some output due to the error being returned.

**Review by Itxaka — APPROVED** [review]

**Comment by davidcassany** [comment]
> I would like to investigate more what happens when we register and things go wrong (right now I think the operator gets stuck with no logs).

As far as I understood the client side gets stuck in this infinite loop https://github.com/rancher/elemental-operator/blob/f655a803422577db1b8a96e486efebacb59565e2/cmd/operator/register/root.go#L132-L149

However this should popup some error messages.

**Pull request merged**